plugins/nf-validation/src/main/nextflow/validation/SchemaValidator.groovy

```diff
@@ -184,16 +184,25 @@ class SchemaValidator extends PluginExtensionPoint {
         def String fileType = SamplesheetConverter.getFileType(samplesheetFile)
         def String delimiter = fileType == "csv" ? "," : fileType == "tsv" ? "\t" : null
         def List<Map<String,String>> fileContent
-        def List<Map<String,String>> fileContentCasted = []
         def Boolean s3PathCheck = params.validationS3PathCheck ? params.validationS3PathCheck : false
+        def Map types = variableTypes(schemaFile.toString(), baseDir)
+        def Boolean containsHeader = !(types.keySet().size() == 1 && types.keySet()[0] == "")
+
+        if(!containsHeader){
+            types = ["empty": types[""]]
+        }
         if(fileType == "yaml"){
-            fileContent = new Yaml().load((samplesheetFile.text))
+            fileContent = new Yaml().load((samplesheetFile.text)).collect {
+                if(containsHeader) {
+                    return it as Map
+                }
+                return ["empty": it] as Map
+            }
```

Comment:
Is the identical code in `SamplesheetConverter:castToType` still required?

Reply:
This isn't in that function? The main issue here is that we need an array of `Map` values for the validation. This is a little hacky workaround for samplesheets that have no header

Reply:
Line 73 of SamplesheetConverter.groovy no?

Reply:
I figure if we are pre-converting here we didn't need it there? But maybe I'm not following the flow properly

Reply:
Do you mean the code in `SamplesheetConverter.convertToList()`?. 
This code here is for the validation, `convertToList` is creating the Nextflow channel, and we don't pass `fileContent` or `fileContentCasted` to this function, they are independent. If we want to reduce code, I would rather remove the validation from `fromSamplesheet()`, but this is opening the discussion about where to validate a sample sheet again and out of topic so don't mind me :D

Reply:
Yes sorry I mixed up `castToType` with `convertToList`! 

Code deduplication was the goal, but agree it is out of scope for this PR/patch release if it's not just a case of deleting what's there

Reply:
Yeah code deduplication should really happen at one point but not here :p